test/services/prompter/database-prompts.js

```diff
@@ -0,0 +1,768 @@
+      nameHandlerStub = sinon.stub(databasePrompts, 'handleName');
+      schemaHandlerStub = sinon.stub(databasePrompts, 'handleSchema');
+      hostNameHandlerStub = sinon.stub(databasePrompts, 'handleHostname');
+      portHandlerStub = sinon.stub(databasePrompts, 'handlePort');
+      userHandlerStub = sinon.stub(databasePrompts, 'handleUser');
+      passwordHandlerStub = sinon.stub(databasePrompts, 'handlePassword');
+      sslHandlerStub = sinon.stub(databasePrompts, 'handleSsl');
+      mongoSrvHandlerStub = sinon.stub(databasePrompts, 'handleMongodbSrv');
+      await databasePrompts.handlePrompts();
+    });
+
+    after(() => {
+      connectionUrlHandlerStub.restore();
+      dialectHandlerStub.restore();
+      nameHandlerStub.restore();
+      schemaHandlerStub.restore();
+      hostNameHandlerStub.restore();
+      portHandlerStub.restore();
+      userHandlerStub.restore();
+      passwordHandlerStub.restore();
+      sslHandlerStub.restore();
+      mongoSrvHandlerStub.restore();
+      resetParams();
+    });
+
+    it('should handle the connection url', () => {
+      expect(connectionUrlHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle the dialect', () => {
+      expect(dialectHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle the name', () => {
+      expect(nameHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle the schema', () => {
+      expect(schemaHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle the port', () => {
+      expect(portHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle the user', () => {
+      expect(userHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle the password', () => {
+      expect(passwordHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle ssl usage', () => {
+      expect(sslHandlerStub.calledOnce).to.equal(true);
+    });
+
+    it('should handle mongodb srv usage', () => {
+      expect(mongoSrvHandlerStub.calledOnce).to.equal(true);
+    });
+  });
+
+  describe('Handling connection url : ', () => {
+    describe('When the dbConnectionUrl option is requested', () => {
+      describe('and the dbConnectionUrl has already been passed in', () => {
+        describe('and the dbConnectionUrl is valid', () => {
+          before(() => {
+            requests.push('dbConnectionUrl');
+          });
+
+          after(() => {
+            resetParams();
+          });
+
+          it('should add the dbConnectionUrl to the configuration', async () => {
+            const databasePrompts = new DatabasePrompts(program, envConfig, prompts, requests);
+
+            program.connectionUrl = 'postgres://username:password@host:port/database';
+
+            await databasePrompts.handleConnectionUrl();
+
+            expect(envConfig.dbConnectionUrl).to.equal(program.connectionUrl);
```

Comment:
Make sure it equals `postgres://username:password@host:port/database` as the `program.connectionUrl` could have been change inside the prompter ;)